app/services/file_locator.rb

```diff
@@ -134,7 +134,18 @@ def reader
     when 's3'
       S3File.new(uri).object.get.body
     when 'file'
-      File.open(location,'r')
+      File.open(location, 'r')
+    else
+      open_uri
+    end
+  end
+
+  def magic_bytes
+    case uri.scheme
+    when 's3'
+      S3File.new(uri).object.get(range: '0-16').body
+    when 'file'
+      File.open(location, 'r')
     else
       open_uri
```

Comment:
I think it would be good if these other cases also returned the first 16 bytes too.  Some quick testing locally shows a speedup for `'file'` when working with large files and fetching just the first 16 bytes.  I didn't see any difference with `open_uri.read(16)` and https://test-videos.co.uk/vids/bigbuckbunny/mp4/h264/1080/Big_Buck_Bunny_1080_10s_30MB.mp4.

```suggestion
      File.read(location, 16)
    else
      open_uri.read(16)
```